src/simularium/VisDataCache.ts

```diff
@@ -36,14 +39,38 @@ class VisDataCache {
     public changeSettings(options: {
         maxSize?: number;
         cacheEnabled?: boolean;
+        onCacheUpdate?: (log: CacheLog) => void;
     }): void {
-        const { maxSize, cacheEnabled } = options;
+        const { maxSize, cacheEnabled, onCacheUpdate } = options;
         if (cacheEnabled !== undefined) {
             this._cacheEnabled = cacheEnabled;
         }
         if (maxSize !== undefined) {
             this._maxSize = maxSize;
         }
+        if (onCacheUpdate !== undefined) {
+            this.logCacheUpdate = onCacheUpdate;
+        }
+        this.onCacheUpdate();
+    }
+
+    public onCacheUpdate(): void {
+        if (this.logCacheUpdate) {
+            this.logCacheUpdate({
+                size: this.size,
+                framesInCache: this.getListOfCachedFrameNumbers(),
+            });
+        }
```

Comment:
I believe TS lets you shrink this to

```suggestion
        this.logCacheUpdate?.({
            size: this.size,
            framesInCache: this.getListOfCachedFrameNumbers(),
        });
```

...if you want to